### PR TITLE
Dark mode

### PR DESCRIPTION
**Pull request by leshihak**

### CODE REVIEW

File: src/components/NoteItem/NoteItem.scss [hunk 1]
```diff
@@ -47,3 +47,25 @@
         padding: 10px;
     }
 }
+
+.container-note--dark {
```
Comment (vitaliyoliynykk) [inline src/components/NoteItem/NoteItem.scss:51]: Why do you copy such properties as paddins, cursor, transition etc. When you only need to change color?

File: src/components/NoteItem/NoteItem.tsx [hunk 2]
```diff
@@ -1,36 +1,60 @@
...
+    const closeModal = (): void => {
+        setIsOpen(false);
+    };
+
     const handleRemoveNoteItem = (id: string): void => {
+        closeModal();
         deleteNoteItem(id);
     };
 
+    const removeModalTitle = 'Do you really want to remove a Note?';
```
Comment (vitaliyoliynykk) [inline src/components/NoteItem/NoteItem.tsx:33]: Move this to separate file NoteItem.constants

File: src/components/NoteItem/NoteItem.tsx [hunk 3]
```diff
@@ -1,36 +1,60 @@
...
+
+    let noteClass = 'container-note';
```
Comment (vitaliyoliynykk) [inline src/components/NoteItem/NoteItem.tsx:35]: it looks like spaghetti code, consider using classname, or at least move it to separate function

File: src/components/NoteItem/NoteItem.tsx [hunk 4]
```diff
@@ -1,36 +1,60 @@
...
 const NoteItem = ({
     note,
     deleteNoteItem,
     isActive,
+    darkMode,
 }: {
     note: Note;
     deleteNoteItem: (id: string) => void;
     isActive: boolean;
+    darkMode: boolean;
```
Comment (vitaliyoliynykk) [inline src/components/NoteItem/NoteItem.tsx:16]: consider rename it "isDarkMode"

File: src/components/NoteItemsList/NoteItemsList.tsx [hunk 5]
```diff
@@ -10,11 +9,13 @@ const NoteItemsList = ({
     removeNoteItem,
     selectNoteItem,
     activeNoteIdProp,
+    darkMode,
 }: {
     arrayOfNotes: Note[];
     activeNoteIdProp?: string;
     removeNoteItem: (id: string) => void;
     selectNoteItem: (note: Note) => void;
+    darkMode: boolean;
```
Comment (vitaliyoliynykk) [inline src/components/NoteItemsList/NoteItemsList.tsx:18]: rename to isDarkMode in the whole project

File: src/components/RemoveModal/RemoveModal.scss [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    }
+    &--title {
+        font-size: 30px;
+    }
+    &--btns {
+        width: 100%;
+        display: flex;
+        justify-content: space-evenly;
+    }
+    &--btn {
```
Comment (vitaliyoliynykk) [inline src/components/RemoveModal/RemoveModal.scss:40]: '--' is modifier, not element. Lets use official methodology http://getbem.com/

File: src/components/RemoveModal/RemoveModal.scss [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        }
+        &-light {
+            @extend .container-modal__block--btn-dark;
+            background-color: white;
+            color: black;
+        }
+    }
+}
+
+.container-modal__block--dark {
```
Comment (vitaliyoliynykk) [inline src/components/RemoveModal/RemoveModal.scss:67]: here you use "--dark" as modifier, and in code above as element. This is inconsistent

File: src/components/TextEditor/TextEditor.scss [hunk 8]
```diff
@@ -52,3 +55,31 @@
...
+        transition: linear 0.2s;
+    }
+    &__notes {
+        &_title {
+            @extend .container-editor__notes_title;
+            background-color: black;
+            color: white;
+            transition: linear 0.2s;
+        }
+        &_description {
```
Comment (vitaliyoliynykk) [inline src/components/TextEditor/TextEditor.scss:81]: you dont need this

**Review by vitaliyoliynykk — CHANGES_REQUESTED** [review]

**Comment by leshihak** [comment]
And watch Notes component for removing notes (bug)

File: src/components/Loader/Loader.scss [hunk 9]
```diff
@@ -1,6 +1,9 @@
 .loader-component {
     display: inline-block;
-    position: relative;
+    position: absolute;
```
Comment (vitaliyoliynykk) [inline src/components/Loader/Loader.scss:3]: you shouldnt set absolute position to component. Instead wrap it into div wherever you want to use it, and set absolute position to div

File: src/containers/Notes/Notes.tsx [hunk 10]
```diff
@@ -38,10 +40,10 @@ class Notes extends React.Component<{}, NotesState> {
             .then((snapshot) => {
                 if (snapshot.val()) {
                     const notes: Note[] = Object.values(snapshot.val());
-                    this.setState({ ...this.state, notes, activeNote: notes[FIRST_ELEMENT] });
+                    this.setState({ ...this.state, notes, activeNote: notes[FIRST_ELEMENT], loading: false });
                 } else {
                     const note = getDefaultNote();
-                    this.setState({ ...this.state, notes: [note], activeNote: note });
+                    this.setState({ ...this.state, notes: [note], activeNote: note, loading: true });
                 }
```
Comment (vitaliyoliynykk) [inline src/containers/Notes/Notes.tsx:47]: add "catch" block to set loading to false if there is an error

**Review by vitaliyoliynykk — CHANGES_REQUESTED** [review]

**Pull request merged**